tests/sentry/models/test_releaseprojectenvironment.py

```diff
@@ -83,3 +92,68 @@ def test_no_update_too_close(self):
         )
         assert release_project_env.first_seen == self.datetime_now
         assert release_project_env.last_seen == self.datetime_now
+
+    @receivers_raise_on_send()
+    @patch.object(ReleaseProjectEnvironmentManager, "subscribe_project_to_alert_rule")
+    def test_post_save_subscribes_project_to_alert_rule_if_created(
+        self, mock_subscribe_project_to_alert_rule
+    ):
+        ReleaseProjectEnvironment.get_or_create(
+            project=self.project,
+            release=self.release,
+            environment=self.environment,
+            datetime=self.datetime_now,
+        )
+
+        assert mock_subscribe_project_to_alert_rule.call_count == 1
+
+    @patch(
```

Comment:
question about sentry testing philosophy... i've seen mocking sparingly and mostly checks for the db to have the conditions completed afterwards. is there a general recommendation?

(I personally like mocking out external logic like this because it improves test performance so much.. but the trade-off is testing implementation a bit more explicitly over the logic of the function)

Reply:
mmmm 🤔 

I know we tend to like full e2e tests in our unit tests....
Personally i'm also in favor of testing the individual units and mocking the relations... IMO makes it easier to more clearly understand whats going on and whats expected?

that being said - i have the non patched tests below as well 😂